src/Symfony/Component/Form/DataAccessorInterface.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form;
+
+/**
+ * Writes and reads values to/from an object or array bound to a form.
+ *
+ * @author Yonel Ceruto <yonelceruto@gmail.com>
+ */
+interface DataAccessorInterface
+{
+    /**
+     * Returns the value at the end of the property of the object graph.
+     *
+     * @param object|array  $objectOrArray The object or array to traverse
+     * @param FormInterface $form          The {@link FormInterface()} instance to check
+     *
+     * @return mixed The value at the end of the property
+     *
+     * @throws Exception\AccessException If unable to read from the given form data
+     */
+    public function getValue($objectOrArray, FormInterface $form);
+
+    /**
+     * Sets the value at the end of the property of the object graph.
+     *
+     * @param object|array  $objectOrArray The object or array to modify
+     * @param mixed         $value         The value to set at the end of the object graph
+     * @param FormInterface $form          The {@link FormInterface()} instance to check
+     *
+     * @throws Exception\AccessException If unable to write the given value
+     */
+    public function setValue(&$objectOrArray, $value, FormInterface $form): void;
```

Comment:
can't the method return the modified value instead of using a ref?

Reply:
Yes, it can. But it is not mandatory to return a new value here and I would like to keep this signature quite similar to the current `DataMapperInterface::mapFormsToData()` method for consistency.